lib/collection/src/shards/replica_set/backoff.rs

```diff
@@ -0,0 +1,74 @@
+use std::cmp;
+use std::collections::HashMap;
+use std::time::{Duration, Instant};
+
+use crate::shards::shard::PeerId;
+
+#[derive(Clone, Debug, Default)]
+pub struct Registry {
```

Comment:
Not sure about the name, suggestions welcome. Don't want to call it `NotifyPeerFailureBackoff`, though, because the type itself is _kinda_ generic.

E.g., we can add `abort_shard_transfer: backoff::Registry` to the `ShardReplicaSet` and use it to rate-limit `abort_shard_transfer` calls independent of `notify_peer_failure` calls.

Reply:
What about something like `RetryLimiter`?

I find `Registry` a bit took ambiguous for something that specifically throttles/limits retry calls for a list of peers.